src/core/lib/security/authorization/audit_logging.h

```diff
@@ -56,12 +56,16 @@ class AuditLoggerRegistry {
   static void TestOnlyResetRegistry();
 
  private:
-  absl::StatusOr<AuditLoggerFactory*> GetAuditLoggerFactory(
-      absl::string_view name);
+  // TODO(lwge): Add built-in logger registrations once avaialble.
+  AuditLoggerRegistry() = default;
+
+  static Mutex mu;
```

Comment:
Note that the style guide dictates that static and global variables must be trivially destructible:

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

I think this means that we'll need both of these static variables to be pointers, and we can dynamically allocate the value at initialization time.

Reply:
> should we only construct the registry when it's being used?

I'd be fine with that change if you want to make it, but I don't think it's strictly necessary.  We're not allocating a huge amount of memory here.

Reply:
Sounds good. I'll leave it as is then.